src/highlight/HighlightAnnotations.tsx

```diff
@@ -88,20 +94,44 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
             })),
         });
         setStatus(CreatorStatus.staged);
+    };
+
+    const handlePromote = (): void => {
+        stageSelection();
 
         setIsPromoting(true);
     };
 
+    React.useEffect(() => {
+        if (!isSelecting) {
+            return;
+        }
+
+        setStaged(null);
+        setStatus(CreatorStatus.init);
+    }, [isSelecting]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    React.useEffect(() => {
+        if (!isCreating || !selection) {
+            return;
+        }
+
+        stageSelection();
+    }, [isCreating, selection]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    // Prevent mousedown and mouseup events that occur within the PopupReply from propagating the HighlightListener.
+    const popupReplyRef = usePreventEventPropagationRef<HTMLDivElement>('mousedown', 'mouseup');
```

Comment:
It seems like we're jumping through a lot of hoops to support the mouseup/down handling on `annotatedEl` in HighlightListener. Could those handlers live in a per-page React component, similar to the RegionCreator?

Reply:
As we discussed, the problem is that `RegionCreator` can handle mouseup/down events on an arbitrary layer but highlights depends on textLayer events and given the way the layers are organized currently, `HighlightAnnotations` layer is a sibling to the textLayer so having a per page React component support the mouseup/down logic might not be doable. Let's keep this for now and we can add an item to the todo list to see if there's a better way to handle this